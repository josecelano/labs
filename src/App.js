/**
 * Set up redux store and react-redux Provider in this file
 */
import React from 'react';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import appState from './reducer';
import TodoList from './TodoList';

const store = createStore(appState);

const App = (props) => {
    return (
        <Provider store={store}><TodoList/></Provider>
    )
}

export default App;
