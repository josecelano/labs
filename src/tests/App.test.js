/**
 * Tests for the App component
 */
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

describe('<App />', () => {
    it('should provide an store to todo list', () => {
        const store = {};
        const tree = renderer
            .create(<App store={ store }/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})