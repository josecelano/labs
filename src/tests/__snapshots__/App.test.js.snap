exports[`<App /> should provide an store to todo list 1`] = `
<div>
  <ul>
    <li>
      <label>
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox" />
        Get to Las Palmas 🌴
      </label>
    </li>
    <li>
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox" />
        Learn the basics of React 👩‍🎓
      </label>
    </li>
    <li>
      <label>
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox" />
        Go surfing 🏄
      </label>
    </li>
  </ul>
  <form
    onSubmit={[Function]}>
    <input
      onChange={[Function]}
      type="text"
      value={undefined} />
  </form>
</div>
`;
